Sources/CryptoKit/Utilities/CryptoData.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+public enum CryptoData {
+    case data(Data)
+    case string(String)
+    case staticString(StaticString)
+    case array(Array<UInt8>)
+    case arraySlice(ArraySlice<UInt8>)
+    
+    public func string() -> String {
```

Comment:
Is there a reason that this and `hexEncodedString` are methods instead of computed properties?

Reply:
@calebkleveter `hexEncodedString()` is a method in Vapor 3, I just ported that over. For `string()`, just `.string` could conflict with the enum case.

I do wonder if I should find a way to make them computed properties, though. Does Swift style guide have any comments on this?

Reply:
Well, this compiles and runs just fine:

```swift
enum CData {
    case string(String)
    
    var string: String {
        switch self {
        case let .string(str): return str
        }
    }
}
```

So the compiler doesn't see a collision (unless it was added in Swift 5. I used 4.2.1).

Reply:
The Swift style guide doesn't appear to say anything about computed properties vs methods, but I think it would be argued that you should use computed properties under the [Uniform access principal](https://en.wikipedia.org/wiki/Uniform_access_principle)<sup>1</sup>

<sup>1</sup><sub>Courtesy of [SoftwareEngineering SE](https://softwareengineering.stackexchange.com/questions/304077/swift-functions-vs-computed-properties)</sub>

Reply:
That's a good point. I just asked jw for his opinion on this and he brought up a good counter point. `string` and `hexEncodedString` are relatively expensive operations (`O(n)`). Usually when someone accesses a property, they expect the runtime to be closer to `O(1)`. Making this a method instead of a computed var could nudge people toward calling it more conservatively which would improve performance. 

Another thing jw mentioned was moving this to a `String` initializer. i.e.,

```swift
 let hash = String(hexRepresenation: SHA1.hash("foo"))
```

Reply:
I like the idea for the initializer 👍